project/SparkBuild.scala

```diff
@@ -426,8 +426,10 @@ object TestSettings {
     fork := true,
     // Setting SPARK_DIST_CLASSPATH is a simple way to make sure any child processes
     // launched by the tests have access to the correct test-time classpath.
-    envVars in Test += ("SPARK_DIST_CLASSPATH" ->
-      (fullClasspath in Test).value.files.map(_.getAbsolutePath).mkString(":").stripSuffix(":")),
+    envVars in Test ++= Map(
+      "SPARK_DIST_CLASSPATH" -> 
+        (fullClasspath in Test).value.files.map(_.getAbsolutePath).mkString(":").stripSuffix(":"),
+      "JAVA_HOME" -> sys.props("java.home")),
```

Comment:
If `JAVA_HOME` was actually set by the user who invoked SBT (or Maven), then I think this change (and the Maven change) will cause us to ignore their preferred JRE in favor of the JRE used to launch the build tool, which might be confusing.